### PR TITLE
Adding plotting, fitting and results table into FDA

### LINKED ISSUE #25987: Add fitting and results table to FDA
Want to add the ability to fit and create results tables in FDA. 

Will need to change:
https://github.com/mantidproject/mantid/blob/master/scripts/Muon/GUI/Common/contexts/muon_context.py#L225

might be useful:
https://github.com/mantidproject/mantid/blob/master/scripts/Muon/GUI/Common/contexts/muon_context.py#L36

https://github.com/mantidproject/mantid/blob/master/scripts/Muon/GUI/FrequencyDomainAnalysis/MaxEnt/maxent_presenter_new.py#L153

### PR DESCRIPTION
**Pull request by AnthonyLim23**
I have:
 - added plotting of the modulus of the freq spectrum to FDA
- added fitting into FDA
- added results table tab into FDA

The unit tests will be added as part of #26037

**To test:**
Open Muon Analysis 2 and make sure it works as before. My changes should not have altered it.

Open frequency domain analysis 
Load some data
Use the GUI.
Things to watch out for:
 - does the plot of freq update correctly?
 - does the default fitting selection make sense?
 - can you fit to the freq spec and make a results table?
 - does it work with PhaseQuad?
 - can you still do time domain analysis?

Fixes #25987  

---
Does not need to be in release notes as it will be covered in a later issue/PR
#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/GUI/Common/contexts/muon_context.py [hunk 1]
```diff
@@ -234,9 +253,15 @@ def get_names_of_workspaces_to_fit(self, runs='', group_and_pair='', phasequad=F
             for run in run_list:
                 run_string = run_list_to_string(run)
                 phasequad_names += self.phase_context.get_phase_quad(self.data_context.instrument, run_string)
-
+        print(group_names, "group names")
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/contexts/muon_context.py]: need to delete

**Comment by AnthonyLim23** [comment]
fix muon analysis 2 ...

**Comment by Harrietbrown** [comment]
Whenever I load any run I get an error: ``Unknown property search object Output UnNormWorkspace``
It doesn't seem to interfere with the loading.

**Comment by AnthonyLim23** [comment]
is it a pop up or in the logger?

**Comment by Harrietbrown** [comment]
Pop up.

**Comment by AnthonyLim23** [comment]
the amount of data generated is annoying me, so I am adding some more filtering

**Comment by AnthonyLim23** [comment]
that should fix the selection being annoying

File: scripts/Muon/GUI/Common/contexts/muon_context.py [hunk 2]
```diff
@@ -32,12 +34,25 @@ def __init__(self, muon_data_context=None, muon_gui_context=None,
...
-        self.ads_observer = MuonContextADSObserver(self.remove_workspace_by_name, self.clear_context)
+        self._frequency_context = frequency_context
 
-        self.gui_context.update({'DeadTimeSource': 'None', 'LastGoodDataFromFile': True, 'selected_group_pair': ''})
+        self.ads_observer = MuonContextADSObserver(
+            self.remove_workspace_by_name,
+            self.clear_context)
+
+        self.gui_context.update(
+            {'DeadTimeSource': 'None',
```
Comment (Harrietbrown) [inline scripts/Muon/GUI/Common/contexts/muon_context.py]: Can these all be in line with each other?

Comment (AnthonyLim23) [reply scripts/Muon/GUI/Common/contexts/muon_context.py]: this was autopep 8 as flake8 does not like lines to get too long.

Comment (Harrietbrown) [reply scripts/Muon/GUI/Common/contexts/muon_context.py]: Sorry I wasn't being clear, I mean can the all the elements of the dictionary have the same indentation. Lines 45 and 46 are indented to line up with the bracket at the end of line 43, but 44 is only indented one tab out from line 43.

Comment (AnthonyLim23) [reply scripts/Muon/GUI/Common/contexts/muon_context.py]: flake8 will catch those sort of things

**Review by Harrietbrown — COMMENTED** [review]
Bug 1: I havn't been able to reproduce it, but during testing I somehow managed to get a phase table into the data selection in the fitting tab. If you can think of any way that would be possible It would be good to fix it. 

Bug 2: Rebinning after fitting data causes an error
1. Open frequency domain analysis.
1. Load MSUR 62260.
1. Fit an Abragam
1. Go back to home tab
1. Set Rebin --> fixed --> steps = 1
1. We get an error and the plot scaled
This not in muon analysis 2 depsite it not actualy using any FDA exclusive functions

**Comment by AnthonyLim23** [comment]
I cant reproduce either of those. The closest I can get is (in FDA):
load some data, do a fit and then go back and rebin
The plot will change and have an extended range (0-32, instead of 1-15).

**Comment by AnthonyLim23** [comment]
fixes:

Open FDA load 62260-2
Set group to top
Set the plot type to freq Re
Do FFT's of 6226n for Re and 62260 for Im
There should only be 3 lines on the plot (previously it would add the old lines again)
Go to fitting and there should only be 3 items in the combo box (previously it picked up each one twice)

Open FDA load 62260-2
Set group to top
Change the group to fwd there should be 3 lines ( previously it kept the top lines too)

Check MaxEnt also gets added to plots and workspace selector correctly 

Open muon analysis 2 load some data
Change the plot type to counts

**Comment by AnthonyLim23** [comment]
that should be all done now. Also fixed:

Open FDA and load some data
Set the plot type to freq Re
Set the group to top
Go to transform and do a FFT
Should get no errors

You should be able to see plots update too when fitting function has changed.

File: scripts/Muon/GUI/FrequencyDomainAnalysis/frequency_context.py [hunk 3]
```diff
@@ -5,115 +5,94 @@
...
+    @property
+    def window_title(self):
+        return "Frequency Domain Analysis"
+
+    def add_maxEnt(self, run, ws_freq):
+        self._maxEnt_freq[ws_freq.name()] = MaxEnt(run, ws_freq)
+
+    @property
+    def maxEnt_freq(self):
+        return [maxEnt for maxEnt in list(self._maxEnt_freq.keys())]
```
Comment (Mantid-Matthew) [inline scripts/Muon/GUI/FrequencyDomainAnalysis/frequency_context.py]: would just list(self._maxEnt_freq.keys()) this do the same here?

File: scripts/Muon/GUI/FrequencyDomainAnalysis/frequency_context.py [hunk 4]
```diff
@@ -5,115 +5,94 @@
...
-        # if period data look for the first period
-        if mantid.AnalysisDataService.doesExist("MuonAnalysis_1"):
-            tmpWS = mantid.AnalysisDataService.retrieve("MuonAnalysis_1")
-            return True, tmpWS
-            # if its not period data
-        elif mantid.AnalysisDataService.doesExist("MuonAnalysis"):
-            tmpWS = mantid.AnalysisDataService.retrieve("MuonAnalysis")
-            return True, tmpWS
+    def FFT_freq(self):
+        return [FFT for FFT in list(self._FFT_freq.keys())]
```
Comment (Mantid-Matthew) [inline scripts/Muon/GUI/FrequencyDomainAnalysis/frequency_context.py]: would just list(self._FFT_freq.keys()) achieve the same here?

File: scripts/test/Muon/frequency_domain_context_test.py [hunk 5]
```diff
@@ -45,10 +43,8 @@ def tearDown(self):
         AnalysisDataService.clear()
 
     def test_get_detectors_excluded_from_default_grouping_tables_gets_correct_groups_for_CHRONUS(self):
-        result = self.frequency_context.get_detectors_excluded_from_default_grouping_tables()
-
-        self.assertEqual(result, [256, 425])
+        pass
```
Comment (Mantid-Matthew) [inline scripts/test/Muon/frequency_domain_context_test.py]: If this test is no longer needed can we just remove it?

File: scripts/Muon/GUI/FrequencyDomainAnalysis/frequency_context.py (hunk 3 above)
Comment (AnthonyLim23) [reply scripts/Muon/GUI/FrequencyDomainAnalysis/frequency_context.py]: yes

**Review by Harrietbrown — DISMISSED** [review]
Everything looks good, I couldn't find any way to break it and everything functions as it should.

**Comment by NickDraper** [comment]
@Anthony could you fix the conflicting files please

**Comment by AnthonyLim23** [comment]
@Mantid-Matthew  is probably better placed, as he has been pushing this through for me.

**Pull request merged**